modules/jaggery-apps/app-publisher-web/apis/v1/lifecycle_api_router.jag

```diff
@@ -21,553 +21,805 @@
-      var artifact = artifactManager.get(id);
-      var checkListItems = artifactManager.getCheckListItemNames(artifact);
-      var msg = {};
-      msg['status'] = 'ok';
-      msg['checkListItems'] = checkListItems;
-      print(msg);
-    } catch (e) {
-      log.debug('The check list names were not retrieved for artifact: ' + id + ' .The following exception was thrown: ' + e);
-      response.sendError(404, 'The check list item names could not be retrieved for artifact: ' + id + ' .Please refer to the server log.');
-    }
-  });
 
-/*
-  Gets the checked or unchecked state of a check list item
-*/
-  routeManager.register('GET', 'publisher', '/publisher/api/lifecycle/checklistitem/{index}/{type}/{id}', function(context) {
-    try {
-      log.debug('Entered checklistitem/index/type/id ');
-      var checkListItemIndex = context.params.index;
-      var shortName = context.params.type;
-      var id = context.params.id;
-
-      var artifactManager = rxtManager.getArtifactManager(shortName);
-
-      var artifact = artifactManager.get(id);
-      var isChecked = artifactManager.isItemChecked(checkListItemIndex, artifact);
-      var msg = {};
-      msg['status'] = 'ok';
-      msg['index'] = checkListItemIndex;
-      msg['isChecked'] = isChecked;
-      print(msg);
-    } catch (e) {
-      log.debug('The check list item index state could not be  retrieved for aritfact: ' + id + ' .The following exception was thrown: ' + e);
-      response.sendError(404, 'The check list item state could not be retrieved for artifact: ' + id + ' .Please refer to the server log.');
-    }
-  });
-
-routeManager.register('GET', 'publisher', '/publisher/api/lifecycle/information/history/{type}/{id}/{version}', function(context) {
-    try {
-      var shortName = context.params.type;
-      var id = context.params.id;
-      var artifactManager = rxtManager.getArtifactManager(shortName);
-      var artifact = artifactManager.get(id);
-      var historyRes = artifactManager.getLifecycleHistory(artifact);
-      var historyComments = require('/modules/comment.js').getLifecycleComments(artifactManager, artifact);
-      var json = {};
-
-      if (historyRes != undefined) {
-        var xml = new XML(historyRes.content);
-        json = utility.xml.convertE4XtoJSON(xml);
-      }
-      print(json);
-   }catch(e){
-       log.debug('The lifecycle history for artifact: '+id+' could not be found.The following exception was thrown: '+e);
-       response.sendError(404,'The lifecycle history details could not be retrieved for artifact: '+id+'.Pleas refer to the server log.');
-   }
-});
-
-var securityModule=require('/modules/security/security.manager.js').securityManagementModule();
-
-var sm=securityModule.cached();
+	/*
+	 The function returns the lifecycle structure of a given asset
+	 */
+	function obtainLifeCycle(type) {
+		CommonUtil = Packages.org.wso2.carbon.governance.lcm.util.CommonUtil;
```

Comment:
Is there any reason to put this object in global context? 
CommonUtil -> commonUtil